prusti-encoder/src/encoders/mir_pure.rs

```diff
@@ -1283,44 +1219,36 @@ pub fn encode_place_element<'vir, 'enc, T: TaskEncoder>(
             assert!(place_ty.variant_index.is_none());
             match place_ty.ty.kind() {
                 TyKind::Adt(adt, _) if adt.is_box() => {
-                    let e_ty = deps
-                        .require_ref::<RustTyPredicatesEnc>(place_ty.ty)
+                    let e_ty_impure = deps
+                        .require_ref::<TyImpureEnc>(place_ty.ty)
                         .unwrap();
-                    let struct_like = e_ty
-                        .generic_predicate
+                    let struct_like = e_ty_impure
                         .expect_variant_opt(place_ty.variant_index);
-                    let proj = struct_like.snap_data.field_access[0];
-                    let proj_app = proj.call()(expr);
+                    let e_ty_pure = deps
+                        .require_local::<TyPureEnc>(place_ty.ty)
+                        .unwrap();
+                    let proj = e_ty_pure.get_variant_opt(place_ty.variant_index).field(abi::FieldIdx::ZERO);
+                    let proj_app = proj.read(expr);
                     let place_ref =
-                        place_ref.map(|pr| struct_like.ref_to_field_refs[0].call()(pr, &[]));
+                        place_ref.map(|pr| struct_like.field(abi::FieldIdx::ZERO, pr));
```

Comment:
Could we have a nicer accessor for the box field, instead of assuming the contents are in the zeroth field?

Reply:
Probably, but currently all the encoders treat `Box` as a one-field struct. I would change this once we have proper unsafe code support (e.g. making `Box` less builtin (except for the `mir::ProjectionElem::Deref`) by adding extern_spec annotations to the definition)